temporal-sdk/src/main/java/io/temporal/internal/worker/Poller.java

```diff
@@ -206,9 +206,11 @@ private class PollLoopTask implements Runnable {
       this.task = task;
       this.pollBackoffThrottler =
           new BackoffThrottler(
```

Comment:
Technically the resource exhausted retry policy could be completely different from the "normal" retry policy.
I'd instead use two different retryer instances but as long as this is all internal and we're free to customize this behavior as needed later this is fine.